backend/de.metas.purchasecandidate.base/src/main/java/de/metas/purchasecandidate/material/event/PurchaseCandidateAdvisedEventCreator.java

```diff
@@ -70,18 +81,33 @@ public Optional<PurchaseCandidateAdvisedEvent> createPurchaseAdvisedEvent(
 			return Optional.empty();
 		}
 
-		final I_PP_Product_Planning productPlanning = mrpContext.getProductPlanning();
-
-		final PurchaseCandidateAdvisedEvent event = PurchaseCandidateAdvisedEvent
-				.builder()
+		final PurchaseCandidateAdvisedEvent.PurchaseCandidateAdvisedEventBuilder event = PurchaseCandidateAdvisedEvent.builder()
 				.eventDescriptor(EventDescriptor.ofEventDescriptor(supplyRequiredDescriptor.getEventDescriptor()))
-				.supplyRequiredDescriptor(supplyRequiredDescriptor)
 				.directlyCreatePurchaseCandidate(productPlanning.isCreatePlan())
 				.productPlanningId(productPlanning.getPP_Product_Planning_ID())
-				.vendorId(defaultVendorProductInfo.get().getVendorId().getRepoId())
-				.build();
+				.vendorId(defaultVendorProductInfo.get().getVendorId().getRepoId());
+
+		if(productPlanning.isLotForLot())
+		{
+			final BigDecimal fullDemandQty = supplyRequiredDescriptor.getFullDemandQty();
```

Comment:
so, afaiu we have `supplyRequiredDescriptor.fullDemandQty` as our "lot4lot-qty" and `supplyRequiredDescriptormaterialDescriptor.quantity` as our "not-lot4lot-qty", right?

This needs to be clearer IMO (even more so in case i got it wrong by looking at the code). So, please rename the fields in the  `SupplyRequiredDescriptor`, or if that'S not practical, add javadoc to the fields, to make clearer which is which.

Reply:
`supplyRequiredDescriptor.materialDescriptor.quantity` gets updated in case of LotForLot with fullDemandQty, because it's the Qty used later on (added javadocs)